src/main/java/org/folio/tag/repository/packages/PackageRepositoryImpl.java

```diff
@@ -75,11 +76,22 @@ public PackageRepositoryImpl(Vertx vertx, IdParser idParser) {
 
   @Override
   public CompletableFuture<List<PackageId>> getPackageIdsByTagName(List<String> tags, int page, int count, String tenantId) {
+    return getPackageIdsByTagAndIdPrefix(tags, "", page, count, tenantId);
+  }
+
+  @Override
+  public CompletableFuture<List<PackageId>> getPackageIdsByTagNameAndProvider(List<String> tags, String providerId, int page, int count, String tenant) {
+    return getPackageIdsByTagAndIdPrefix(tags, providerId, page, count, tenant);
+  }
+
+  private CompletableFuture<List<PackageId>> getPackageIdsByTagAndIdPrefix(List<String> tags, String providerId, int page, int count, String tenantId) {
     int offset = (page - 1) * count;
 
     JsonArray parameters = new JsonArray();
     tags.forEach(parameters::add);
+    String likeExpression = providerId + "%";
     parameters
```

Comment:
When filtering by providerId -- should like be for `providerId-` -- would not want to get provider id 18 and 181 results for example when searching for provider id 18.

Reply:
Thanks